asf_public_discourse_home_decarbonisation/pipeline/bert_topic_analysis/title_topic_analysis.py

```diff
@@ -0,0 +1,159 @@
+- [optional] MIN_TOPIC_SIZE is the minimum size of a topic. Defaults to 100.
+"""
+
+# Package imports
+import pandas as pd
+import argparse
+import logging
+from datetime import datetime
+
+logger = logging.getLogger(__name__)
+
+# Local imports
+from asf_public_discourse_home_decarbonisation.utils.text_processing_utils import (
+    process_abbreviations,
+)
+from asf_public_discourse_home_decarbonisation import S3_BUCKET
+from asf_public_discourse_home_decarbonisation.getters.public_discourse_getters import (
+    read_public_discourse_data,
+)
+from asf_public_discourse_home_decarbonisation.utils.topic_analysis_utils import (
+    get_outputs_from_topic_model,
+)
+from asf_public_discourse_home_decarbonisation.utils.topic_analysis_utils import (
+    update_topics_with_duplicates,
+    update_docs_with_duplicates,
+    topic_model_definition,
+)
+
+
+def parse_arguments() -> argparse.Namespace:
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--source",
+        help="`mse` or `buildhub`",
+        required=True,
+    )
+    parser.add_argument(
+        "--reduce_outliers_to_zero",
+        help="True to reduce outliers to zero. Defaults to False",
+        default=False,
+        type=bool,
+    )
+    parser.add_argument(
+        "--start_date",
+        help="Analysis start date in the format YYYY-MM-DD. Default to None (all data)",
+        default=None,
+        type=str,
+    )
+    parser.add_argument(
+        "--end_date",
+        help="Analysis end date in the format YYYY-MM-DD. Defaults to None (all data)",
+        default=None,
+        type=str,
+    )
+    parser.add_argument(
+        "--min_topic_size",
+        help="Minimum topic size. Defaults to 100",
+        default=100,
+        type=int,
+    )
+    return parser.parse_args()
+
+
+if __name__ == "__main__":
+    # Dealing with user defined arguments
+    args = parse_arguments()
+    source = args.source
+    reduce_outliers_to_zero = args.reduce_outliers_to_zero
+    start_date = args.start_date
+    end_date = args.end_date
+    min_topic_size = args.min_topic_size
+
+    forum_data = read_public_discourse_data(source=source)
+
+    # prep date/time data
+    if source == "buildhub":
+        forum_data.rename(columns={"url": "id", "date": "datetime"}, inplace=True)
+    forum_data["datetime"] = pd.to_datetime(forum_data["datetime"])
+    forum_data["date"] = forum_data["datetime"].dt.date
+    forum_data["year"] = forum_data["datetime"].dt.year
+
+    # filter by date
+    if start_date is not None:
+        forum_data = forum_data[
+            forum_data["date"] >= datetime.strptime(start_date, "%Y-%m-%d").date()
+        ]
+    if end_date is not None:
+        forum_data = forum_data[
+            forum_data["date"] <= datetime.strptime(end_date, "%Y-%m-%d").date()
+        ]
+
+    forum_data["title"] = forum_data["title"].apply(process_abbreviations)
+    forum_data["title"] = forum_data["title"].replace("hp", "heat pump")
```

Comment:
What's the reason for this happening here rather than the process_abbreviations step before? Might be me being dim haha.

Reply:
no, it's a good question - I can clarify it in the documentation.

So when processing the text, you can't replace hp with heat pump because it will replace stuff like php with heat pump when you have a link ending in .php

Titles don't have links so you can now replace HP with heat pump - but only because we're only collecting the subset related to home heating. If we had collected wider posts we couldn't do it, because they might have mentioned HP as in the computer/printer/technology brand.